File: app/src/main/java/com/worldventures/dreamtrips/modules/friends/model/RequestHeaderModel.java
package com.worldventures.dreamtrips.modules.friends.model;

public class RequestHeaderModel {

   boolean advanced;
   String name;
   private int count;

   public RequestHeaderModel(String name) {
      this(name, false);
   }

   public RequestHeaderModel(String name, boolean advanced) {
      this.name = name;
      this.advanced = advanced;
   }

   public boolean isAdvanced() {
      return advanced;
   }

   public String getName() {
      return name;
   }

   public int getCount() {
      return count;
   }

   public void setCount(int count) {
      this.count = count;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/users/base/model/RequestHeaderModel.kt
package com.worldventures.dreamtrips.social.service.users.base.model

class RequestHeaderModel(val name: String?, val isAdvanced: Boolean) {

   var count: Int = 0

   constructor(name: String?) : this(name, false)

   override fun equals(other: Any?) = (other as? RequestHeaderModel)?.let {
      name == it.name && count == it.count && isAdvanced == it.isAdvanced
   } ?: false

   override fun hashCode() = name?.hashCode() ?: 0 + count
}
